src/components/Modal/index.tsx

```diff
@@ -1,69 +1,199 @@
+  & {
+    align-items: center;
+    flex-direction: column;
+    display: flex;
+    overflow-y: scroll;
+  }
+
+  .paper {
+    position: relative;
+    top: 68px;
+    width: 500px;
+    border-radius: 8px;
+    background-color: #ffffff;
+    box-shadow: 0 0 5px 0 rgba(74, 85, 121, 0.5);
+    display: flex;
+    flex-direction: column;
+
+    &:focus {
+      outline: none;
+    }
+
+    // TODO: replace class-based styles by params
+    &.receive-modal {
+      height: auto;
+      max-width: calc(100% - 130px);
+      min-height: 544px;
+      overflow: hidden;
+    }
+
+    &.bigger-modal-window {
+      width: 775px;
+      height: auto;
+    }
+
+    &.smaller-modal-window {
+      height: auto;
+    }
+
+    &.modal {
+      height: auto;
+      max-width: calc(100% - 130px);
+      overflow: hidden;
+    }
+  }
+`
 
 interface GnoModalProps {
   children: ReactNode
   description: string
   // type copied from Material-UI Modal's `close` prop
-  handleClose?: {
-    bivarianceHack(event: Record<string, unknown>, reason: 'backdropClick' | 'escapeKeyDown'): void
-  }['bivarianceHack']
-  modalClassName?: string
+  handleClose?: (event: Record<string, unknown>, reason: 'backdropClick' | 'escapeKeyDown') => void
   open: boolean
   paperClassName?: string
   title: string
 }
 
-const GnoModal = ({
-  children,
-  description,
-  handleClose,
-  modalClassName,
-  open,
-  paperClassName,
-  title,
-}: GnoModalProps): ReactElement => {
-  const classes = useStyles()
-
+const GnoModal = ({ children, description, handleClose, open, paperClassName, title }: GnoModalProps): ReactElement => {
   return (
-    <Modal
-      aria-describedby={description}
-      aria-labelledby={title}
-      className={cn(classes.root, modalClassName)}
-      onClose={handleClose}
-      open={open}
-    >
-      <div className={cn(classes.paper, paperClassName, 'classpep')}>{children}</div>
-    </Modal>
+    <ModalStyled aria-describedby={description} aria-labelledby={title} onClose={handleClose} open={open}>
+      <div className={cn('paper', paperClassName)}>{children}</div>
+    </ModalStyled>
   )
 }
 
 export default GnoModal
+
+/*****************/
+/* Generic Modal */
+/*****************/
+
+/*** Header ***/
+const HeaderSection = styled.div`
+  display: flex;
+  padding: 16px 24px;
+  border-bottom: 2px solid ${({ theme }) => theme.colors.separator};
+
+  .title-icon {
+    width: 20px;
+    margin-right: 10px;
+  }
+
+  .title-note {
+    margin-left: 12px;
+  }
+
+  .close-button {
+    align-self: flex-end;
+    background: none;
+    border: none;
+    padding: 5px;
+    width: 26px;
+    height: 26px;
+
+    span {
+      margin-right: 0;
+    }
+
+    :hover {
+      background: ${({ theme }) => theme.colors.separator};
+      border-radius: 16px;
+      cursor: pointer;
+    }
+  }
+`
+
+const HeaderTitle = styled(Title)`
+  flex-basis: 100%;
+`
+
+interface HeaderProps {
+  title: string
+  iconUrl?: string
+  titleNote?: string
```

Comment:
I think if we are handling `iconUrl` and `titleNote`, we should also handle the step.
`step?: {current: number, steps: number}`.

Otherwise, this should only be a wrapper over a children.

Reply:
Yes. Actually, the `iconUrl` prop was added for compatibility with the sApps modal.

But, as you stated in the early meeting, better go with something generic for the title and the close button position and then add a wrapper on top of it for each case.